packages/react/src/components/navigation/progress/Stepper/index.tsx

```diff
@@ -151,9 +152,13 @@ export const Step = memo(function Step({
           <Flex flex="1" />
         )}
       </Item.Spacer>
-      <StepText inactive={inactive} errored={errored} variant="small">
-        {label}
-      </StepText>
+      {typeof label === "string" ? (
+        <StepText inactive={inactive} errored={errored} variant="small">
+          {label}
+        </StepText>
+      ) : (
+        label
```

Comment:
The issue with passing a plain react node as a label is that it would be impossible to style depending on the step and its state (active/inactive/errored).

Reply:
indeed, I'll change this then!